src/errors.ts

```diff
@@ -34,6 +38,18 @@ class ConnectionError extends WidgetError {
   }
 }
 
+export class SwapError extends WidgetError {
+  constructor(config: WidgetErrorConfig) {
+    super({
+      ...config,
```

Comment:
nit: I prefer this syntax:

    super({
      header: DEFAULT_DISMISSABLE_ERROR_HEADER,
      action: DEFAULT_DISMISSABLE_ERROR_ACTION,
      ...config,
    })